jdaviz/configs/specviz/helper.py

```diff
@@ -64,15 +69,104 @@ def get_spectral_regions(self):
             Mapping from the names of the subsets to the subsets expressed
             as `specutils.SpectralRegion` objects.
         """
-        regions = self.app.get_subsets_from_viewer('spectrum-viewer')
+        regions = self.app.get_subsets_from_viewer("spectrum-viewer")
 
         spec_regs = {}
 
         for name, reg in regions.items():
-            unit = reg.meta.get('spectral_axis_unit', u.Unit('Angstrom'))
-            spec_reg = SpectralRegion.from_center(reg.center.x * unit,
-                                                  reg.width * unit)
+            unit = reg.meta.get("spectral_axis_unit", u.Unit("Angstrom"))
+            spec_reg = SpectralRegion.from_center(reg.center.x * unit, reg.width * unit)
 
             spec_regs[name] = spec_reg
 
         return spec_regs
+
+    def x_limits(self, x_min=None, x_max=None):
+        """Sets the limits of the x-axis
+
+        :param x_min: The lower bound of the axis. Can also be a
+            Specutils SpectralRegion
+        :param x_max: The upper bound of the axis
+        """
+        scale = self.app.get_viewer("spectrum-viewer").scale_x
+        if not x_min and not x_max:
+            return scale
+        else:
+            # Retrieve the spectral axis
+            ref_index = self.app.get_viewer(
+                "spectrum-viewer"
+            ).state.reference_data.label
+            spectral_axis = self.get_spectra(ref_index).get(ref_index).spectral_axis
+            self._set_scale(scale, spectral_axis, x_min, x_max)
```

Comment:
We don't strictly need an `else` clause here since the method will return before it gets evaluated.

```suggestion

        # Retrieve the spectral axis
        ref_index = self.app.get_viewer(
            "spectrum-viewer"
        ).state.reference_data.label
        spectral_axis = self.get_spectra(ref_index).get(ref_index).spectral_axis
        self._set_scale(scale, spectral_axis, x_min, x_max)
```